From a0d50f542a8452b99284815f515e604b7ee3e03a Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Mon, 15 Apr 2013 21:02:33 +0200
Subject: [PATCH] xbps-install: revert fa40928026bd.

If a package is already installed and it's being installed without -f (force),
xbps-install must return EEXIST.

xbps-src really expects this to ignore duplicated build dependencies.
---
 bin/xbps-install/transaction.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/bin/xbps-install/transaction.c b/bin/xbps-install/transaction.c
index 6afcad4..cc25e98 100644
--- bin/xbps-install/transaction.c
+++ bin/xbps-install/transaction.c
@@ -218,7 +218,6 @@ install_new_pkg(struct xbps_handle *xhp, const char *pkg, bool reinstall)
 	if ((rv = xbps_transaction_install_pkg(xhp, pkg, reinstall)) != 0) {
 		if (rv == EEXIST) {
 			printf("Package `%s' already installed.\n", pkg);
-			rv = 0;
 		} else if (rv == ENOENT) {
 			fprintf(stderr, "Unable to locate '%s' in "
 			    "repository pool.\n", pkg);
-- 
1.8.2

